core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -852,6 +863,33 @@ public boolean isResponseContentDeferred() {
         return responseContentDeferred;
     }
 
+    @Override
+    public HttpHeaders responseTrailers() {
+        ensureAvailability(RESPONSE_END);
+        return responseTrailers;
+    }
+
+    @Override
+    public void responseTrailers(HttpHeaders responseTrailers) {
+        if (isAvailabilityAlreadyUpdated(RESPONSE_END) || responseTrailers.isEmpty()) {
+            return;
+        }
+
+        final HttpHeaders trailers = this.responseTrailers;
+        final HttpHeaders toSet;
+        if (trailers == EMPTY_HEADERS) {
+            final HttpHeaders newHeaders = new DefaultHttpHeaders();
+            if (responseTrailersUpdater.compareAndSet(this, EMPTY_HEADERS, newHeaders)) {
+                toSet = newHeaders;
+            } else {
+                toSet = this.responseTrailers;
+            }
+        } else {
+            toSet = trailers;
+        }
+        toSet.setAll(responseTrailers);
```

Comment:
We could keep the list of received trailers and merge them at once in `endResponse`, because otherwise we may be vulnerable to a client which sends many small trailers.